config/clusters/2i2c/anu.values.yaml

```diff
@@ -34,10 +34,21 @@ jupyterhub:
         authenticator_class: cilogon
       CILogonOAuthenticator:
         oauth_callback_url: https://anu.pilot.2i2c.cloud/hub/oauth_callback
-        username_claim: "email"
         admin_users:
           - matthew.mckay@anu.edu.au
-        # We do not define allowed_users here since only usernames matching these domains will be allowed to login into the hub.
+        # Only show the option to login with Google and ANU
+        shown_idps:
+          - https://idp2.anu.edu.au/idp/shibboleth
+          - https://accounts.google.com/o/oauth2/auth
         allowed_idps:
-          - 2i2c.org
-          - anu.edu.au
+          http://google.com/accounts/o8/id:
+            username_derivation:
+              username_claim: "email"
+          https://idp2.anu.edu.au/idp/shibboleth:
+            username_derivation:
+            username_claim: "email"
+      Authenticator:
+        # We only want 2i2c and anu users to sign up
+        username_pattern: '^(.+@2i2c\.org|.+@anu\.edu\.au|deployment-service-check)$'
+        # Delete any prior existing users in the db that don't pass username_pattern
+        # delete_invalid_users: true
```

Comment:
```suggestion
        delete_invalid_users: true
```

Without this, hub logs contain errors about previous usernames being in the database and not matching the regex. For example, from when we were using our @gmail.com addresses instead of our 2i2c ones. I believe it's a good rule of thumb to delete users that are not valid anymore?

Error from logs is:
```
[E 2022-06-09 15:16:08.554 JupyterHub app:1884] Error adding user georgiana.dolocan@gmail.com already in db
    Traceback (most recent call last):
      File "/usr/local/lib/python3.8/dist-packages/jupyterhub/app.py", line 1880, in init_users
        f = self.authenticator.add_user(user)
      File "/usr/local/lib/python3.8/dist-packages/jupyterhub/auth.py", line 619, in add_user
        raise ValueError("Invalid username: %s" % user.name)
    ValueError: Invalid username: georgiana.dolocan@gmail.com
    
[W 2022-06-09 15:16:08.554 JupyterHub app:1891] 
    You can set
        c.Authenticator.delete_invalid_users = True
    to automatically delete users from the Hub database that no longer pass
    Authenticator validation,
    such as when user accounts are deleted from the external system
    without notifying JupyterHub.
```

Reply:
+1